index.js

```diff
@@ -66,6 +64,33 @@ app.post('/newgame', function(request, response){
 
 });
 
-app.listen(app.get('port'), function() {
-  console.log("Node app is running at localhost:" + app.get('port'))
-});
\ No newline at end of file
+var server = http.createServer(app).listen(app.get('port'), function(){
+  console.log('Express server listening on port ' + app.get('port'));
+});
+
+var io = require('socket.io').listen(server);
+
+io.on('connection', function (socket) {
+    
+    socket.on('join', function (gameid) {
+        var room = roommaster.findRoom(parseInt(gameid));
+        if(room != null) {
+            socket.join(gameid) // Add this socket to the room with this gameid
+            //TODO: Make a new Player and add them to the game, also send them all the info for the game they're in
+            io.to(gameid).emit('initRoom', room.toString())
+        }
+        else {
+            io.to(gameid).emit('roomDeleted')
+        }
+    });
+
+    // Handle an example event with example data
+    socket.on('exampleClientEvent', function(exampleData) {
+        for(var room in socket.rooms) { // Loop through every room this connection belongs to

```

Comment:
I think there might be a better way to do this but for now if a connection belongs to a room, then any change they send the server will be treated as if it affects that room